x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openai/embeddings/OpenAiEmbeddingsModel.java

```diff
@@ -84,8 +84,11 @@ public ExecutableAction accept(OpenAiActionVisitor creator, Map<String, Object>
     }
 
     public OpenAiEmbeddingsModel overrideWith(Map<String, Object> taskSettings) {
-        var requestTaskSettings = OpenAiEmbeddingsRequestTaskSettings.fromMap(taskSettings);
+        if (taskSettings == null || taskSettings.isEmpty()) {
```

Comment:
I don't really understand this behavior.

If I call `model.overrideWith(Map.of())`, I expect the settings to be cleared, not to be preserved. Why this unintuitiuve behavior?